python/pyspark/daemon.py

```diff
@@ -54,8 +54,9 @@ def worker(sock, authenticated):
     # Read the socket using fdopen instead of socket.makefile() because the latter
     # seems to be very slow; note that we need to dup() the file descriptor because
     # otherwise writes also cause a seek that makes us miss data on the read side.
-    infile = os.fdopen(os.dup(sock.fileno()), "rb", 65536)
-    outfile = os.fdopen(os.dup(sock.fileno()), "wb", 65536)
+    buffer_size = int(os.environ.get("SPARK_BUFFER_SIZE", 65536))
```

Comment:
is `int(os.environ.get("SPARK_BUFFER_SIZE", 65536))` going to return something sensible if `SPARK_BUFFER_SIZE` is set to something crazy like `-1`?

Reply:
I think so .... 

```python
>>> int("-1")
-1
>>> int("a")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: invalid literal for int() with base 10: 'a'
```

Reply:
We can add sanity check to `spark.buffer.size`? At least preventing negative one?

Reply:
Yea. Let me wait a bit to decide if I add 4 > or negative ones.

Reply:
Let me add a negative checking for now.